src/pkg/packager/prepare.go

```diff
@@ -356,6 +377,28 @@ func (p *Packager) processUnstructuredImages(resource *unstructured.Unstructured
 	return matchedImages, maybeImages, nil
 }
 
+func (p *Packager) findWhyResources(resources []*unstructured.Unstructured, whyImage, componentName, resourceName string, isChart bool) []string {
+	output := []string{}
```

Comment:
Rather than return the output as a string slice, we can return a string slice of found `resourceNames`. So if the why image is found in a resource, append the `resourceName` to the slice.

We can print the resource YAML inside of this function as well rather than construct the output outside of this function

Reply:
It's something I did earlier but reverted to constructing the output outside as the componentDefinition or the output for `zarf dev find-images` was also being constructed outside. Will go with your suggestion here. I noticed that the output is more readable when we print the resource yaml from inside of this function.